documents/coredocument.go

```diff
@@ -619,3 +615,12 @@ func populateVersions(cd *coredocumentpb.CoreDocument, prevCD *coredocumentpb.Co
 	}
 	return nil
 }
+
+// GetTestCoreDocWithReset must only be used by tests for manipulations. It gets the embedded coredoc protobuf.
+// All calls to this function will cause a regeneration of salts next time for precise-proof trees.
+func (cd *CoreDocument) GetTestCoreDocWithReset() *coredocumentpb.CoreDocument {
```

Comment:
does this have to be in this file? it doesn't access any private fields, right?

Reply:
yes, but this prevents tests from breaking all over the place, when coredoc changes. We can modify this function to help with the tests as well see all the test use cases.

Reply:
I don't agree. If tests are broken, we should fix them accordingly, let's not add test related code into production. This function returns the pointer of the Core document. This means I can mutate the document. Move this function into test file or to file with build tags